edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/TaskExTest.cs

```diff
@@ -237,6 +238,118 @@ public async Task MayThrow_LetsExceptionThrough_When_NotExpected()
                     });
         }
 
+        [Fact]
+        [Unit]
+        public async Task SelectAsyncNoLimitTest()
+        {
+            TaskCompletionSource<int> completionSource = new TaskCompletionSource<int>();
+            int numStarted = 0;
+            Task<int> Func(int _)
+            {
+                numStarted++;
+                return completionSource.Task;
+            }
+
+            var result = Enumerable.Range(0, 10).SelectAsync(Func);
+            await Task.Delay(100);
```

Comment:
is there a better way to test rather than to use Delay? Timers are not super reliable. if the only thing you care is the completion - why not just await result and check that all 10 completed?

Another idea is that you can spin-wait till numStarted == N or timeout.

Reply:
The point of the delay is just to give the task time to complete. There is no behavior dependent on specific timing, I'm setting the status of the tasks using the completion source. If I don't have the short delay between setting the task completion and checking result's completion, result doesn't have time to run. 

I don't want to await the result, b/c one of the things that's being tested is that result completes. If I await result, a failure case of the test (result never completing) leads to awaiting forever (or the 300 second test kill delay)

Reply:
Yes, that make sense. What I meant is - even though 100ms should be enough - there is no guarantee that your tasks will complete in that time, and test will be false negative. I'm wondering if we can have more deterministic approach here.